scenario_player/tasks/blockchain.py

```diff
@@ -122,16 +133,10 @@ def __init__(
     ) -> None:
         super().__init__(runner, config, parent)
 
-        required_keys = ["contract_name", "event_name", "num_events"]
-        all_required_options_provided = all(key in config.keys() for key in required_keys)
-        if not all_required_options_provided:
-            raise ScenarioError(
-                "Not all required keys provided. Required: " + ", ".join(required_keys)
-            )
-
+        verify_config(config, required_keys=["contract_name", "event_name"])
```

Comment:
Nitpick: I'd rather use a tuple instead of a list here since we don't need mutability for `required_keys`.